addons/product/product.py

```diff
@@ -784,8 +784,9 @@ def write(self, cr, uid, ids, vals, context=None):
     def copy(self, cr, uid, id, default=None, context=None):
         if default is None:
             default = {}
-        template = self.browse(cr, uid, id, context=context)
-        default['name'] = _("%s (copy)") % (template['name'])
```

Comment:
could use `default.setdefault('name',  _("%s (copy)") % (template['name']))` to minimise the diff